### PR TITLE
Don't save progress if video genre is music

### PR DESCRIPTION
**Pull request by vr10t**
---
Don't save progress if video genre is music closes 
---

**Important note**
We may remove your pull request if you do not use this provided PR template correctly.

**Pull Request Type**
Please select what type of pull request this is:
- [x] Bugfix
- [ ] Feature Implementation

**Related issue**
 #1522

**Description**
There is no need to save watch progress for music. This makes it so that if the video genre is music, it sets the watch progress to 0 instead of the db value.

**Screenshots (if appropriate)**
Please add before and after screenshots if there is a visible change.

**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested?
Please describe shortly how you tested it and whether there are any ramifications remaining. 
I tested it for a few days with no issues on Invidious API, but I am unable to test on Local API since it doesn't work for me. I don't think there should be any issues, but if someone else can test it I would appreciate it.
**Desktop (please complete the following information):**
 - OS: [Manjaro]
 - OS Version: [5.10.117-1]
 - FreeTube version: [0.17.0]

**Additional context**

### CODE REVIEW

**Comment by vr10t** [comment]
Tested local API as well and everything seems fine.

**Review by absidue — CHANGES_REQUESTED** [review]
Hi @vr10t, thank you for your pull request.

Could you please add a setting for this change? There are definitely situations in which someone might want to save the progress for videos in the "Music" category and continue where they left off.
Here are some examples: festival recordings, playlists that were uploaded as a single video and live sets, just to name a few.

**Comment by vr10t** [comment]
@absidue  Thanks for the review. I've updated it with settings now.

File: src/renderer/components/privacy-settings/privacy-settings.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ export default Vue.extend({
     saveWatchedProgress: function () {
       return this.$store.getters.getSaveWatchedProgress
     },
+    saveWatchedProgressForMusic: function () {
+      return this.$store.getters.getSaveWatchedProgress
```
Comment (absidue) [inline src/renderer/components/privacy-settings/privacy-settings.js:38]:
```suggestion
      return this.$store.getters.getSaveWatchedProgressForMusic
```

**Review by absidue — CHANGES_REQUESTED** [review]
Could you please fix that small typo? The rest of the code looks good.

**Comment by vr10t** [comment]
> Could you please fix that small typo? The rest of the code looks good.
done, thanks

File: src/renderer/views/Watch/Watch.js [hunk 2]
```diff
@@ -757,14 +761,21 @@ export default Vue.extend({
       if (this.rememberHistory && !this.isUpcoming && !this.isLoading && !this.isLive) {
         const player = this.$refs.videoPlayer.player
 
-        if (player !== null && this.saveWatchedProgress) {
+        if (player !== null && this.saveWatchedProgress && this.saveWatchedProgressForMusic) {
```
Comment (ChunkyProgrammer) [inline src/renderer/views/Watch/Watch.js:764]: If you want to save watch progress but not watch progress for music then wouldnt this not allow saving watch progress?

Comment (vr10t) [reply src/renderer/views/Watch/Watch.js:764]: Yes you are right, I fixed it now.

File: src/renderer/views/Watch/Watch.js [hunk 3]
```diff
@@ -293,7 +296,7 @@ export default Vue.extend({
             channelName: this.channelName,
             channelId: this.channelId
           })
-
```
Comment (PikachuEXE) [inline src/renderer/views/Watch/Watch.js:296]: Keep the blank line?

File: src/renderer/views/Watch/Watch.js [hunk 4]
```diff
@@ -758,10 +762,19 @@ export default Vue.extend({
         const player = this.$refs.videoPlayer.player
 
         if (player !== null && this.saveWatchedProgress) {
-          const currentTime = this.getWatchedProgress()
+          if (this.saveWatchedProgressForMusic || (!this.saveWatchedProgressForMusic && this.genre !== 'Music')) {
```
Comment (PikachuEXE) [inline src/renderer/views/Watch/Watch.js:765]:
```suggestion
          if (this.saveWatchedProgressForMusic || this.genre !== 'Music') {
```

File: src/renderer/views/Watch/Watch.js [hunk 5]
```diff
@@ -758,10 +762,19 @@ export default Vue.extend({
...
+              watchProgress: currentTime
+            }
+            this.updateWatchProgress(payload)
+          }
+        }
+        if (this.genre === 'Music' && !this.saveWatchedProgressForMusic) {
           const payload = {
             videoId: this.videoId,
-            watchProgress: currentTime
+            watchProgress: 0
```
Comment (PikachuEXE) [inline src/renderer/views/Watch/Watch.js:777]: There is no need to update the watch progress to 0 explicitly
(That shouldn't be part of this PR even it's a new intended behaviour)

File: src/renderer/views/Watch/Watch.js [hunk 6]
```diff
@@ -758,10 +762,19 @@ export default Vue.extend({
...
 
         if (player !== null && this.saveWatchedProgress) {
-          const currentTime = this.getWatchedProgress()
+          if (this.saveWatchedProgressForMusic || (!this.saveWatchedProgressForMusic && this.genre !== 'Music')) {
+            const currentTime = this.getWatchedProgress()
+            const payload = {
+              videoId: this.videoId,
+              watchProgress: currentTime
+            }
+            this.updateWatchProgress(payload)
```
Comment (PikachuEXE) [inline src/renderer/views/Watch/Watch.js:771]: Duplicate code
Remove the one below maybe (for setting progress to 0)

**Review by PikachuEXE — CHANGES_REQUESTED** [review]

**Comment by PikachuEXE** [comment]
@vr10t ?

**Comment by ChunkyProgrammer** [comment]
@PikachuEXE @absidue would either of you like to take this PR over? If not I could do it

**Comment by PikachuEXE** [comment]
@ChunkyProgrammer Just do it~
There are only a few changes left

**Comment by PikachuEXE** [comment]
How to find videos with genre "music" ._.

**Comment by ChunkyProgrammer** [comment]
> How to find videos with genre "music" ._.

Here is an example video i found:
https://www.youtube.com/watch?v=rgaTLrZGlk0

I think i might have opened up an issue for adding these details on the watch page in freetube but here it is in newpipe:
 image

**Comment by PikachuEXE** [comment]
If https://www.youtube.com/watch?v=rgaTLrZGlk0 is also "music" then this feature shouldn't be included?
Coz it's not actually music video but "video about music"

**Comment by PikachuEXE** [comment]
I will close this until we have a way to clearly differentiate music videos (but not videos about music) from others
